### PR TITLE
[Pipeline] Replace tests by running a datadog-agent child pipeline

### PR DESCRIPTION
**Pull request by Pythyu**
Since [we’re building system probe in omnibus](https://github.com/DataDog/datadog-agent/pull/19492), datadog-agent-buildimages CI is failing because tests are trying to build the agent without building system-probe, which is now required. Hence we're replacing tests by a datadog-agent child pipeline to fix current and future issues about how the agent is builded.
Full details on [this document](https://docs.google.com/document/d/1udjaYBdw82ZV7aVsJ9ZeFPVzlcDLLgY9c2xr5Zo1Jag/edit).

### CODE REVIEW

File: .gitlab-ci.yml
Comment (chouquette) [inline .gitlab-ci.yml:281]: This isn't used anymore no?

File: .gitlab-ci.yml
Comment (chouquette) [inline .gitlab-ci.yml:408]: I think you can move the `needs` into the `.release` section, which would reduce the duplicated boilerplate

File: .gitlab-ci.yml
Comment (chouquette) [inline .gitlab-ci.yml:250]: Shouldn't we run all kitchen tests for this kind of changes as well? It seems similar in term of risks to changes in omnibus-software

**Comment by chouquette** [comment]
Besides the few questions/remarks, LGTM on principle :)

**Comment by KevinFairise2** [comment]
I have a small question, it looks like the triggered pipeline does not appear in the list of the "Run checks", do we have a way to display them here? I am afraid that we will never know when the triggered pipeline fails and we will always ignore it.

**Comment by chouquette** [comment]
> I have a small question, it looks like the triggered pipeline does not appear in the list of the "Run checks", do we have a way to display them here? I am afraid that we will never know when the triggered pipeline fails and we will always ignore it.

In order to be displayed there, a job *must* be a non trigger job, ie. it must have a `script` section. Its name must also not match one of these patterns: `notify,merge,ng:build,ng:debs,tag,debs,ci-image,after-build,after-test`

File: .gitlab-ci.yml
Comment (Pythyu) [reply .gitlab-ci.yml:250]: previous tests were only testing to build the agent, I'll rather stick to it to not increase to much CI time

**Comment by amenasria** [comment]
**👏 praise:** ‏Can't wait for this to be merged 🙀

**Review by chouquette — APPROVED** [review]
LGTM, having a non-trigger job so we can mark it as required in github can come at a later point in order to unblock people requiring new buildimages

**Pull request merged**
